sdk/typescript/src/cryptography/publickey.ts

```diff
@@ -0,0 +1,111 @@
+// Copyright (c) 2022, Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import BN from 'bn.js';
+import { Buffer } from 'buffer';
+
+/**
+ * Value to be converted into public key
+ */
+export type PublicKeyInitData =
+  | number
+  | string
+  | Buffer
+  | Uint8Array
+  | Array<number>
+  | PublicKeyData;
+
+/**
+ * JSON object representation of PublicKey class
+ */
+export type PublicKeyData = {
+  /** @internal */
+  _bn: BN;
+};
+
+export const PUBLIC_KEY_SIZE = 32;
+
+function isPublicKeyData(value: PublicKeyInitData): value is PublicKeyData {
+  return (value as PublicKeyData)._bn !== undefined;
+}
+
+/**
+ * A public key
+ */
+export class PublicKey {
+  /** @internal */
+  _bn: BN;
```

Comment:
why not private?

Reply:
The class is meant to be compatible with `PublicKeyData`

```
/**
 * JSON object representation of PublicKey class
 */
export type PublicKeyData = {
  /** @internal */
  _bn: BN;
};
```